src/pool/srv_pool.c

```diff
@@ -711,9 +711,9 @@ pool_rsvc_client_complete_rpc(struct rsvc_client *client, const crt_endpoint_t *
  *					replica ranks
  */
 int
-ds_pool_svc_create(const uuid_t pool_uuid, int ntargets, const char *group,
-		   const d_rank_list_t *target_addrs, int ndomains, const uint32_t *domains,
-		   daos_prop_t *prop, d_rank_list_t **svc_addrs)
+ds_pool_svc_dist_create(const uuid_t pool_uuid, int ntargets, const char *group,
```

Comment:
@kccain, the original cases I had in mind is that a PS first performs a series of reconfigurations and then notifies the MS of the result. In this process, there's always a point in time when the MS may have only a proper subset of the PS replicas. If a PoolDestroy comes in at this time, we may leak the PS replicas that the MS doesn't know about. Also, if the PS encounters some errors during the series of reconfigurations, I think it may have created some replicas on ranks that will not be added to the membership until the ranks are picked again in some future series of reconfigurations. Hence, it might be simpler to just let every engine in the pool to try destroying their local PS replica (if any) during PoolDestroy.

The out_creation point is good and I didn't notice it. Are you suggesting we could simplify things further by removing the clean-up ds_pool_svc_stop call in ds_pool_svc_dist_create? I tend to agree, for the less CoRPCs perhaps the better. Should I remove that ds_pool_svc_stop call?

Reply:
>Are you suggesting we could simplify things further by removing the clean-up ds_pool_svc_stop call in ds_pool_svc_dist_create? I tend to agree, for the less CoRPCs perhaps the better. Should I remove that ds_pool_svc_stop call?

In ds_pool_svc_dist_create() there is good (for code reading) symmetry between the ds_rsvc_dist_start() and the error scenario call to ds_rsvc_dist_stop(). If it doesn't cause any regressions I guess the ds_rsvc_dist_stop() call could be removed for its optimization/runtime simplification. But it would be helpful to leave a comment where the call is removed, with a reminder that the error cleanup is being done elsewhere.

Reply:
Sure, I've added cross reference comments. Some manually injected faults have shown that we won't leak replicas when PoolCreate fails.